iree/vm/test/emitc/module_test.cc

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_replace.h"
+#include "iree/base/logging.h"
+#include "iree/base/status.h"
+#include "iree/testing/gtest.h"
+#include "iree/vm/api.h"
+
+// This would be generated together with the functions in the header
+#include "iree/vm/test/emitc/arithmetic_ops.h"
+
+namespace {
+
+typedef iree_status_t (*create_function_t)(iree_allocator_t,
+                                           iree_vm_module_t**);
+
+struct TestParams {
+  std::string module_name;
+  std::string local_name;
+  create_function_t create_function;
+};
+
+struct ModuleDescription {
+  iree_vm_native_module_descriptor_t descriptor;
+  create_function_t create_function;
+};
+
+std::ostream& operator<<(std::ostream& os, const TestParams& params) {
+  return os << absl::StrReplaceAll(params.local_name, {{":", "_"}, {".", "_"}});
+}
+
+std::vector<TestParams> GetModuleTestParams() {
+  std::vector<TestParams> test_params;
+
+  // TODO(simon-camp) get these automatically
+  std::vector<ModuleDescription> modules = {
+      {arithmetic_ops_descriptor_, arithmetic_ops_create}};
+
+  for (int i = 0; i < modules.size(); i++) {
+    iree_vm_native_module_descriptor_t descriptor = modules[i].descriptor;
+    create_function_t function = modules[i].create_function;
+
+    std::string module_name =
+        std::string(descriptor.module_name.data, descriptor.module_name.size);
+
+    for (int i = 0; i < descriptor.export_count; i++) {
```

Comment:
prefer `iree_host_size_t` (or in general the same exact type as the type of the loop bound you are iterating over) - otherwise certain compilers will whine (signed/unsigned mismatch, etc). Here and above.